python/instrumentation/openinference-instrumentation-mistralai/tests/openinference/instrumentation/mistralai/test_instrumentor.py

```diff
@@ -831,9 +933,58 @@ def test_synchronous_streaming_chat_completions_with_tool_call_response_emits_ex
     assert attributes.pop(LLM_TOKEN_COUNT_COMPLETION) == 23
     assert attributes.pop(LLM_TOKEN_COUNT_TOTAL) == 119
     assert attributes.pop(LLM_MODEL_NAME) == "mistral-large-latest"
+    _check_context_attributes(attributes, session_id, user_id, metadata, tags)
     assert attributes == {}  # test should account for all span attributes
 
 
+def _check_context_attributes(
+    attributes: Dict[str, Any],
+    session_id: str,
+    user_id: str,
+    metadata: Dict[str, Any],
+    tags: List[str],
+) -> None:
+    assert attributes.pop(SESSION_ID, None) == session_id
+    assert attributes.pop(USER_ID, None) == user_id
+    attr_metadata = attributes.pop(METADATA, None)
+    assert attr_metadata is not None
+    assert isinstance(attr_metadata, str)  # must be json string
+    metadata_dict = json.loads(attr_metadata)
+    assert metadata_dict == metadata
+    attr_tags = attributes.pop(TAG_TAGS, None)
+    assert attr_tags is not None
+    assert len(attr_tags) == len(tags)
+    assert list(attr_tags) == tags
```

Comment:
it's a best practice i try to follow to avoid abstraction as much as possible in test cases. duplication, especially of something like an assert statement, is very acceptable and even encouraged. the reason being that test cases need to be easy to understand because they need to point you to issues quickly, be stupidly easy to understand, and serve as documentation for other developers. i like the way they put it here: [dry is for code, damp is for tests](https://stackoverflow.com/a/11837973)

Reply:
I agree in general but in this case we're testing, on each case, that the attributes match the fixtures. I do not see the benefit of expanding this until we want to make more complex combined tests. However, if it's a must, I can explode the helper function on each test